### ISSUE TITLE
Add names to editor's controls

### ISSUE THREAD

**Issue by menip** [godotengine/godot#29291]
**Godot version:**
3.1.1

**Issue description:**
Ran into this issue while working on a plugin. Editor controls are for the most part unnamed, making it difficult to find what you want. Here is sample output from `get_editor_interface().get_base_control().print_tree()`:

> .
@@14
@@14/@@15
@@14/@@15/@@54
@@14/@@15/@@54/@@57
@@14/@@15/@@54/@@57/@@56
@@14/@@15/@@54/@@57/@@56/@@55
@@14/@@15/@@54/@@57/@@56/Export
@@14/@@15/@@54/@@57/@@56/Export/@@1711
@@14/@@15/@@54/@@57/@@56/RecentScenes

This goes on for many many lines, with only the occasional control having a relevant name. Note that [Zylann's plugin](https://github.com/Zylann/godot_editor_debugger_plugin) helps somewhat when exploring editor, but does not solve the issue. With this setup, I ended up grabbing the `Play the Project` button via `get_editor_interface().get_base_control().get_child(1).get_child(0).get_child(4).get_child(0)`.
It would of been much cleaner to be able to for example:
`get_editor_interface().get_base_control().get_node("VBox/TopBar/PlayButtons/PlayProjectButton")`.

In short: adding names to editor's controls will make at the least some aspects of plugin dev more straightforward.

**Steps to reproduce:**
Open reproduction project and look at output.

**Minimal reproduction project:**
[New Game Project.zip](https://github.com/godotengine/godot/files/3233718/New.Game.Project.zip)

**Comment by Zylann** [comment]
I think nodes of the editor were not meant to be accessed by name. The whole editor was developped with this assumption, so doing what you did will most of the time be a hack that can potentially break things.

**Comment by TheDuriel** [comment]
`The whole editor was developped with this assumption`

The problem with that, is that the current Plugin APIs are very lackluster when it comes to being able to actually expand the editors functionality. Want a new dock? Sure no problem. Want to remove the asset library button? 

First add a control to the tool bar `add_control_to_container(EditorPlugin.CONTAINER_TOOLBAR, freeze_button)`

Then path from there: `freeze_button.get_parent().get_child(2).get_child(3).visible = false`

Want to force all docks to have a specific minimum size, to avoid the stupid inspector resizing thing?
```	var nuke := Control.new()
	for i in 8:
		add_control_to_dock(i, nuke)
		nuke.get_parent().rect_min_size = Vector2(256, 256)
		yield(get_tree(), "idle_frame")
		remove_control_from_docks(nuke)
	nuke.queue_free()
```

Oh my god.

**Comment by Zylann** [comment]
They could be given names, which means they would be part of the API: they need to be considered, maintained that way, and documented.

**Comment by TheDuriel** [comment]
A good start may be to collect every "button container" for all the editors, expanding the EditorPlugin.CONTAINER_XXX list.

**Comment by menip** [comment]
Adding to container list would be nice. Relative paths more likely to stay same, and as long as "important" nodes are named, their type and hierarchy can be changed in minor version without too much hassle from plugin devs.

**Comment by Calinou** [comment]
I think the main point of doing this would be to improve editor debuggability, which is especially useful for engine contributors.

I'm willing to take a look at naming essential editor controls, but I'm waiting for approval first :slightly_smiling_face:

**Comment by Calinou** [comment]
Closing in favor of https://github.com/godotengine/godot-proposals/issues/1018, as feature proposals are now tracked on the Godot proposals repository.
